test/boost/aws_error_injection_test.cc

```diff
@@ -0,0 +1,280 @@
+
+                                 <Error>
+                                  <Code>InternalError</Code>
+                                  <Message>We encountered an internal error. Please try again.</Message>
+                                  <RequestId>656c76696e6727732072657175657374</RequestId>
+                                  <HostId>Uuag1LuByRx9e6j5Onimru9pO4ZVKnJ2Qz7/C1NPcfTWAtRPfTaOFg==</HostId>
+                                 </Error>)";
+                    case failure_policy::NONRETRYABLE_FAILURE:
+                        assert(!aws::aws_error_map.at("InvalidAction").is_retryable());
+                        return R"(<?xml version="1.0" encoding="UTF-8"?>
+
+                                 <Error>
+                                  <Code>InvalidAction</Code>
+                                  <Message>Something went terribly wrong</Message>
+                                  <RequestId>656c76696e6727732072657175657374</RequestId>
+                                  <HostId>Uuag1LuByRx9e6j5Onimru9pO4ZVKnJ2Qz7/C1NPcfTWAtRPfTaOFg==</HostId>
+                                 </Error>)";
+                    }
+                }
+            }
+            return response;
+        }
+        server& _test_server;
+    };
+    explicit server(failure_policy _test_failure_policy) : test_failure_policy(_test_failure_policy) {
+        handler = std::make_unique<dummy_aws_request_handler>(*this);
+    }
+    future<> start(const std::string& address, uint16_t port) {
+        co_await seastar::async([this, port, address{std::move(address)}] {
+            net::inet_address addr(address);
+            std::cout << "Starting server on " << address << ":" << port << std::endl;
+            http_server.start("test").get();
+
+            http_server.server()
+                .invoke_on_all([this](httpd::http_server& server) {
+                    server._routes.add_default_handler(handler.get());
+                    return make_ready_future<>();
+                })
+                .get();
+
+            http_server.listen(socket_address{addr, port}).handle_exception([](auto ep) { return make_exception_future<>(ep); }).get();
+        });
+    }
+    future<> stop() {
+        co_await http_server.server().stop();
+        co_await http_server.stop();
+    }
+
+private:
+    std::unique_ptr<httpd::handler_base> handler;
+    httpd::http_server_control http_server;
+    std::atomic<failure_policy> test_failure_policy;
+};
+
+static std::function<uint16_t(void)> get_port = [] { return 2012; };
+static std::function<std::string(void)> get_address = [] { return "127.0.0.1"; };
+
+struct unshare_fixture {
+    unshare_fixture() {
+        if (unshare(CLONE_NEWNET) != -1) {
+            std::system("ip l s lo up");
+            BOOST_TEST_MESSAGE("The network namespace was unshared");
+        } else {
+            BOOST_TEST_MESSAGE("Cant unshare network namespace. Randomizing addresses and ports");
+            get_port = []() {
```

Comment:
see https://github.com/scylladb/scylladb/blob/master/test/pylib/host_registry.py. my concern is that this machinery could clash with the one used by `HostRegistry`.